File: Practica 4/Figura.java
public class FiguraGeometrica {
    
    double area(double radio)
    {
        return Math.PI * radio * radio; //circulo
    }
    
    double area(double base, double altura)
    {
        return base * altura; //rectangulo
    }
    
    
    double area(double base, double altura, String figura) {
        if (figura.equalsIgnoreCase("triangulo")) {
            return (base * altura) / 2; // Área del triángulo rectángulo
        }
        return 0; 
    }

    
    double area(double baseMayor, double baseMenor, double altura, String figura) {
        if (figura.equalsIgnoreCase("trapecio")) {
            return ((baseMayor + baseMenor) * altura) / 2; // Área del trapecio
        }
        return 0; 
    }

    
    double area(double lado, String figura) 
    {
        if (figura.equalsIgnoreCase("pentagono")) {
            return (5 * lado * lado) / (4 * Math.tan(Math.PI / 5)); 
        }
        return 0;
    }
    
    public static void main(String[] args) 
    {
        FiguraGeometrica f1 = new FiguraGeometrica();
        FiguraGeometrica f2 = new FiguraGeometrica();
        FiguraGeometrica f3 = new FiguraGeometrica();
        FiguraGeometrica f4 = new FiguraGeometrica();
        FiguraGeometrica f5 = new FiguraGeometrica();
        FiguraGeometrica figura = new FiguraGeometrica();
        
       
        System.out.println("Circulo: " + f1.area(5));
        System.out.println("Rectangulo: " + f2.area(4,6));
        System.out.println("Triangulo Rectangulo: " + figura.area(3, 4, "triangulo")); 
        System.out.println("Trapecio: " + figura.area(3, 5, 4, "trapecio")); 
        System.out.println("Pentagono Regular: " + figura.area(2, "pentagono")); 
    }
    
}

File: Practica 4/Figura.py
import math

class FiguraGeometrica:
    
    def area(self, *args):
        if len(args) == 1:  # Círculo (radio)
            return math.pi * args[0] ** 2
        elif len(args) == 2:
            if isinstance(args[1], str):  # Pentágono (lado, figura)
                lado, figura = args
                if figura.lower() == "pentagono":
                    return (5.0 * lado * lado) / (4 * math.tan(math.pi / 5))
            else:  # Rectángulo (base, altura)
                return args[0] * args[1]
        elif len(args) == 3:  # Triángulo rectángulo (base, altura, figura)
            base, altura, figura = args
            if figura.lower() == "triangulo":
                return (base * altura) / 2
        elif len(args) == 4:  # Trapecio (base mayor, base menor, altura, figura)
            baseMayor, baseMenor, altura, figura = args
            if figura.lower() == "trapecio":
                return ((baseMayor + baseMenor) * altura) / 2
        else:
            raise ValueError("Número de argumentos inválido")


f1 = FiguraGeometrica()
f2 = FiguraGeometrica()
f3 = FiguraGeometrica()
f4 = FiguraGeometrica()
f5 = FiguraGeometrica()
figura = FiguraGeometrica()

print("Círculo:", f1.area(5))  # Radio
print("Rectángulo:", f2.area(4, 6))  # Base y altura
print("Triángulo Rectángulo:", figura.area(3, 4, "triangulo"))  # Base y altura
print("Trapecio:", figura.area(3, 5, 4, "trapecio"))  # Base Mayor, Base Menor, Altura
print("Pentágono Regular:", figura.area(2, "pentagono"))  # Lado
